File: 1. Fundamental-Level/CSharp/03. OOP-November-2015/[HW]/04. Inheritance-And-Abstraction/04. [HW]/01. Human-Student-Worker/Worker.cs
using System;

public class Worker : Human
{
    private double weekSalary;
    private int workHourPerDay;

    public Worker(string firstName, string lastName, double weekSalary, int workHourPerDay)
        : base(firstName, lastName)
    {
        this.WeekSalary = weekSalary;
        this.WorkHoursPerDay = workHourPerDay;
    }

    public double WeekSalary
    {
        get
        {
            return this.weekSalary;
        }
        set
        {
            if (value < 0)
            {
                throw new ArgumentOutOfRangeException("The work hours per day cannot be nagative.");
            }

            this.weekSalary = value;
        }
    }
    public int WorkHoursPerDay
    {
        get
        {
            return this.workHourPerDay;
        }
        set
        {
            if (value < 0)
            {
                throw new ArgumentOutOfRangeException("The work hours per day cannot be nagative.");
            }

            this.workHourPerDay = value;
        }
    }

    public double MoneyPerHour()
    {
        return this.WeekSalary / this.WorkHoursPerDay;
    }

    public override string ToString()
    {
        return string.Format("{0} {1}\nWeek Salary - {2}\nWork hours per day - {3}\n", this.FirstName, this.LastName, this.WeekSalary, this.WorkHoursPerDay);
    }
}

File: 1. Fundamental-Level/Java/OOP-Bacis-June-2016/[EX]/05. Inheritance/src/problem3_Mankind/Worker.java
package problem3_Mankind;

public class Worker extends Human {
    private double weekSalary;
    private double workHoursPerDay;

    public Worker(String firstName, String lastName, double weekSalary, double workHoursPerDay) {
        super(firstName, lastName);
        this.setWeekSalary(weekSalary);
        this.setWorkHoursPerDay(workHoursPerDay);
    }

    private void setWeekSalary(double weekSalary) {
        if (weekSalary < 10) {
            throw new IllegalArgumentException("Expected value mismatch!Argument: weekSalary");
        }

        this.weekSalary = weekSalary;
    }

    private void setWorkHoursPerDay(double workHoursPerDay) {
        if (workHoursPerDay < 1 || workHoursPerDay > 12) {
            throw new IllegalArgumentException("Expected value mismatch!Argument: workHoursPerDay");
        }

        this.workHoursPerDay = workHoursPerDay;
    }

    private double calculateSalaryPerHour() {
        return this.weekSalary / (7 * this.workHoursPerDay);
    }

    @Override
    public String toString() {
        return super.toString() +
                "Week Salary: " + String.format("%.2f", this.weekSalary) + System.lineSeparator() +
                "Hours per day: " + String.format("%.2f", this.workHoursPerDay) + System.lineSeparator() +
                "Salary per hour: " + String.format("%.2f", this.calculateSalaryPerHour()) + System.lineSeparator();
    }
}